lib/home/widgets/route_carousel.dart

```diff
@@ -54,21 +55,48 @@ class _RouteCarouselState extends State<RouteCarousel> {
               ],
             ),
           ),
-          if (widget.routes.isEmpty) widget.emptyWidget,
-          if (widget.routes.isNotEmpty)
-            SizedBox(
+          // if (widget.routes.isEmpty) widget.emptyWidget,
+          // if (widget.routes.isNotEmpty)
+          SizedBox(
               height: 250.0,
-              child: ListView.builder(
-                scrollDirection: Axis.horizontal,
-                itemCount: widget.routes.length,
-                itemBuilder: (BuildContext context, int index) {
-                  return RouteCard(
-                      route: widget.routes[index],
-                      first: index == 0,
-                      last: index == (widget.routes.length - 1));
+              child: StreamBuilder<List<RouteData>>(
+                stream: widget.routes,
+                builder: (BuildContext context,
+                    AsyncSnapshot<List<RouteData>> snapshot) {
+                  if (snapshot.hasError) {
+                    return const Text("Error getting routes.");
+                  } else {
+                    switch (snapshot.connectionState) {
+                      case ConnectionState.none:
+                        return widget.emptyWidget;
+                      case ConnectionState.waiting:
+                        return const SizedBox(
+                            width: 250,
+                            child: Padding(
+                                padding: EdgeInsets.all(75),
+                                child: CircularProgressIndicator()));
```

Comment:
Personally I feel like the CircularProgressIndicator is a bit too big. What do you think? Maybe play around with resizing it and/or wrapping it in a `Center` widget

Reply:
Created a [card](https://nmd2117.atlassian.net/jira/software/projects/RCJ/boards/1?selectedIssue=RCJ-56) to use skeleton loading instead